src/main/java/org/molgenis/vcf/utils/metadata/FieldMetadataServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+package org.molgenis.vcf.utils.metadata;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import htsjdk.variant.vcf.*;
+import org.molgenis.vcf.utils.UnexpectedEnumException;
+import org.molgenis.vcf.utils.model.NumberType;
+import org.molgenis.vcf.utils.model.ValueDescription;
+import org.molgenis.vcf.utils.model.json.JsonFieldMetadata;
+import org.molgenis.vcf.utils.model.json.JsonFieldMetadatas;
+import org.molgenis.vcf.utils.model.json.NestedJsonFieldMetadata;
+import org.molgenis.vcf.utils.model.metadata.FieldMetadata;
+import org.molgenis.vcf.utils.model.metadata.FieldMetadatas;
+import org.molgenis.vcf.utils.model.metadata.NestedFieldMetadata;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.vcf.utils.metadata.FieldType.FORMAT;
+import static org.molgenis.vcf.utils.metadata.FieldType.INFO;
+import static org.molgenis.vcf.utils.metadata.ValueCount.Type.*;
+
+public class FieldMetadataServiceImpl implements FieldMetadataService {
+    private final Map<FieldIdentifier, NestedAttributes> nestedAttributesMap;
+    private final File jsonMetadata;
+
+    public FieldMetadataServiceImpl(Map<FieldIdentifier, NestedAttributes> nestedAttributesMap, File jsonMetadata) {
+        this.nestedAttributesMap = requireNonNull(nestedAttributesMap);
+        this.jsonMetadata = requireNonNull(jsonMetadata);
+    }
+
+    public FieldMetadatas load(VCFHeader vcfHeader) {
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, FieldMetadata> infoMatadata = new HashMap<>();
+        Map<String, FieldMetadata> formatMatadata = new HashMap<>();
+
+        try {
+            JsonFieldMetadatas jsonFieldMetadatas = mapper.readValue(new FileInputStream(jsonMetadata), JsonFieldMetadatas.class);
```

Comment:
use try-with-resources on new FileInputStream(jsonMetadata) to ensure that it is properly closed